File: Data Structures & Algorithms/anagram-groups/submission-6.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String,List<String>> map = new HashMap<>();
        for(String i : strs){
            char[] S=i.toCharArray();
            Arrays.sort(S);
            String s = new String(S);
            if(map.containsKey(s)){
                map.get(s).add(i);
            }
            else{
                List<String> list = new ArrayList<>();
                list.add(i);
                map.put(s,list);
            }
        }
        List<List<String>> l = new ArrayList<>();
        for (String key : map.keySet()){
            l.add(map.get(key));
        }
        return l;
        
    }
}

File: Data Structures & Algorithms/string-encode-and-decode/submission-6.py
class Solution:

    def encode(self, strs: List[str]) -> str:
        s=""
        for i in strs:
            n=len(i)
            s+=str(n)+'#'+i
        return s

    def decode(self, s: str) -> List[str]:
        i=0
        l=[]
        while i<len(s):
            j=i
            while j < len(s) and s[j] !="#":
                j+=1
            length=int(s[i:j])
            l.append(s[j+1:j+1+length])
            i=j+1+length
        return l


